### PR TITLE
Pivotal #ID 174275642: Submission loading strategy

### PR DESCRIPTION
**Pull request by Juan-EBI**
https://www.pivotaltracker.com/story/show/174275642

### CODE REVIEW

File: submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/model/DbSection.kt [hunk 1]
```diff
@@ -15,24 +15,44 @@ import javax.persistence.Id
...
+            Node(ATTRS),
+            Node(FILES, subgraph = "attrs"),
+            Node(SECTS, subgraph = SECTION_FULL_GRAPH)
+        ]),
+    NamedEntityGraph(
+        name = SECTION_SIMPLE_GRAPH,
+        attributeNodes = [
+            Node(ATTRS),
+            Node(SECTS),
+            Node(LINKS, subgraph = "attrs-sub-graph"),
```
Comment (jhoanmanuelms) [inline submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/model/DbSection.kt:43]: Couldn't you just use the same "attrs" subgraph here?

File: submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/SubmissionRepository.kt [hunk 2]
```diff
@@ -42,9 +52,30 @@ class SubmissionRepository(
             .map { it.asSimpleSubmission() }
     }
 
-    private fun getSubmission(accNo: String) =
-        submissionRepository.getByAccNo(accNo) ?: throw SubmissionNotFoundException(accNo)
+    /**
+     * Load submission information strategy used is basically first load submission and then load each section and it
```
Comment (jhoanmanuelms) [inline submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/SubmissionRepository.kt:56]: typo: its subsections

File: submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/SubmissionRepository.kt [hunk 3]
```diff
@@ -42,9 +52,30 @@ class SubmissionRepository(
...
+     * subsections recursively starting from the root section.
+     */
+    private fun submission(accNo: String, version: Int? = null): DbSubmission {
```
Comment (jhoanmanuelms) [inline submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/SubmissionRepository.kt:59]: I think `loadSubmission` would be clearer

File: submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/SubmissionRepository.kt [hunk 4]
```diff
@@ -42,9 +52,30 @@ class SubmissionRepository(
...
+        val submission = findSubmission(accNo, version)
+        submission ?: throw SubmissionNotFoundException(accNo)
+        loadSection(submission.rootSectionId)
+
+        logger.info { "Loaded submission $accNo" }
+        return submission
+    }
+
+    private fun findSubmission(accNo: String, version: Int?): DbSubmission? {
+        return when (version) {
```
Comment (jhoanmanuelms) [inline submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/SubmissionRepository.kt:71]: Are we sure we need this logic? why not just leave the version with a default value?

Comment (Juan-EBI) [reply submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/SubmissionRepository.kt:71]: I notice that was kind of misleading. With default parameter, we do not know if we need a specific version or we need an active one

**Pull request merged**